File: modules/server/eventdata/particles.lua
Particles = {}

if not ServerConfig.Modules.Particles.enabled then
    return
end

function Particles.ProcessEventData(source, data)
	local entityNetId = data[4]
	local entity = NetworkGetEntityFromNetworkId(entityNetId)

	if DoesEntityExist(entity) then
		local owner = NetworkGetEntityOwner(entity)
		if owner ~= source then
			TriggerEvent("icarus:my602oxd71pv", source, "ptFxEvent [C1]", false)
			CancelEvent()
		end
	end

	local scale = tonumber(data[24])
	if scale ~= nil and scale > ServerConfig.Modules.Particles.maxScale then
		TriggerEvent("icarus:my602oxd71pv", source, "ptFxEvent [C2]", false, {fxScale = scale})
		CancelEvent()
	end
end

File: modules/server/eventdata/particles.js
module.exports = (source, data) => {
    if (data.scale > 0 && data.scale > serverConfig.Modules.Particles.maxScale) {
        CancelEvent();
    }
    if (data.isOnEntity) {
        const owner = NetworkGetEntityOwner(NetworkGetEntityFromNetworkId(data.entityNetId));
        if (source == owner) { return; }
        emit("icarus:my602oxd71pv", source, "ptFxEvent [C1]", false, {
            source: source,
            entityNetId: entityNetId
        });
        CancelEvent();
    }
};